modules/oe_translation_cdt/modules/oe_translation_cdt_mock/src/Plugin/ServiceMock/TokenApi.php

```diff
@@ -0,0 +1,45 @@
+<?php
+
+declare(strict_types=1);
+
+namespace Drupal\oe_translation_cdt_mock\Plugin\ServiceMock;
+
+use Drupal\Core\Site\Settings;
+use GuzzleHttp\Psr7\Response;
+use Psr\Http\Message\RequestInterface;
+use Psr\Http\Message\ResponseInterface;
+
+/**
+ * Intercepts token request made to CDT.
+ *
+ * @ServiceMock(
+ *   id = "oe_translation_cdt_token_api",
+ *   label = @Translation("CDT mocked token responses for testing."),
+ *   weight = -1,
+ * )
+ */
+class TokenApi extends ServiceMockBase {
+
+  /**
+   * {@inheritdoc}
+   */
+  protected function getEndpointUrl(): string {
+    return Settings::get('cdt.token_api_endpoint');
+  }
+
+  /**
+   * {@inheritdoc}
+   */
+  protected function needsToken(): bool {
+    return FALSE;
+  }
+
+  /**
+   * {@inheritdoc}
+   */
+  public function getEndpointResponse(RequestInterface $request): ResponseInterface {
+    $this->log('200: Returning the mocked authorization token.', $request);
```

Comment:
Authorize the connection unconditionally, regardless of the username and password.